aea/decision_maker/base.py

```diff
@@ -100,133 +84,44 @@ def update(self, new_status: Status) -> None:
         self._status = new_status
 
 
-class OwnershipState:
+class OwnershipState(ABC):
     """Represent the ownership state of an agent."""
 
-    def __init__(self):
-        """
-        Instantiate an ownership state object.
-
-        :param decision_maker: the decision maker
-        """
-        self._amount_by_currency_id = None  # type: Optional[CurrencyHoldings]
-        self._quantities_by_good_id = None  # type: Optional[GoodHoldings]
-
-    def _set(
-        self,
-        amount_by_currency_id: CurrencyHoldings,
-        quantities_by_good_id: GoodHoldings,
-    ) -> None:
+    @abstractmethod
+    def set(self, **kwargs) -> None:
```

Comment:
Here we have a problem: set should only be accessible to the dm..several ways to address it but non sticks out to me (e.g. could separate it into the Data class and access interface class, only access interface shared with skill context)